src/dsp_tools/xmllib/internal_helpers.py

```diff
@@ -0,0 +1,90 @@
+from __future__ import annotations
+
+import warnings
+from typing import Any
+
+from dsp_tools.models.custom_warnings import DspToolsUserInfo
+from dsp_tools.models.custom_warnings import DspToolsUserWarning
+from dsp_tools.models.exceptions import InputError
+from dsp_tools.xmllib.models.config_options import NewlineReplacement
+from dsp_tools.xmllib.models.config_options import Permissions
+from dsp_tools.xmllib.models.values import Richtext
+from dsp_tools.xmllib.value_checkers import is_string_like
+from dsp_tools.xmllib.value_converters import replace_newlines_with_tags
+
+
+def create_richtext_with_checks(
+    value: str,
+    prop_name: str,
+    permissions: Permissions,
+    comment: str | None,
+    newline_replacement: NewlineReplacement,
+    res_id: str,
+) -> Richtext:
+    """
+    Creates a RichtextValue with checks and optional conversions
+
+    Args:
+        value: richttext value
+        prop_name: name of the property
+        permissions: permissions of the value
+        comment: comment for the value
+        newline_replacement: the replacement for the newlines in the string
+        res_id: id of the calling resource
+
+    Returns:
+        A richtext value
+
+    Raises:
+        Input Error if the input is a dictionary
+    """
+    # Because of the richtext conversions, the input value is cast as a string.
+    # Values such as str(`pd.NA`) result in a non-empy string.
+    # Therefore, a check must occur before the conversion takes place.
```

Comment:
@Nora-Olivia-Ammann Where does a conversion take place? In `replace_newlines_with_tags()`, the input is not converted into a string.